Add new mainframe section in /doc/platforms

gshriver opened the pull request:
This PR adds a new mainframe section in /docs/platforms and contains a gentle introduction to mainframe systems. This new mainframe section is targeted at readers who may be familiar with cloud-native environments and OpenTelemetry, but who may be unfamiliar with legacy mainframe systems.

- [x] I have read and followed the
      [contribution guidelines](https://opentelemetry.io/docs/contributing/),
      including the **First-time contributing?** note.
- [x] This PR includes **AI generated content**, and I have read and conform to
      the
      [Generative AI Contribution Policy](https://github.com/open-telemetry/community/blob/main/policies/genai.md).

- **Preview**: [/docs/platforms/mainframe/](https://deploy-preview-8624--opentelemetry.netlify.app/docs/platforms/mainframe/)

gshriver committed:
Initial commit

gshriver committed:
Remove future pages... section

gshriver committed:
Add Current Status section

gshriver committed:
Correct typo

gshriver committed:
Correct typo

gshriver commented:
Again, sorry for the confusion and extra work.  I see that the CLA is failing.  I have opened an issue with LFX support requesting assistance.  If you want me to close this PR while trying to work this out, I understand.  Thank you.

gshriver committed:
Incorporate feedback from Broadcom internal stakeholders

gshriver committed:
Incorporate more internal feedback

gshriver committed:
Fix SMF hyperlink and other issues

gshriver committed:
Fix capitalization error

cartermp reviewed (commented):
Mostly LGTM

otelbot[bot] committed:
Results from /fix directive

svrnm reviewed (changes requested):
overall LGTM, in a follow up PR we need to add @open-telemetry/sig-mainframe-approvers as component owners

svrnm commented:
@open-telemetry/sig-mainframe-approvers PTAL

gshriver committed:
Update content/en/docs/platforms/mainframe/_index.md

Add link to OpenTelemetry concepts to make it easier for the reader

Co-authored-by: Severin Neumann <severin.neumann@altmuehlnet.de>

gshriver committed:
Add requested Community links

gshriver commented on content/en/docs/platforms/mainframe/_index.md:
```diff
@@ -0,0 +1,167 @@
+
+If you are interested in contributing, see the community and SIG information in
+the OpenTelemetry repositories and website for meeting times and communication
+channels.
```
Thank you.  Added links in commit https://github.com/open-telemetry/opentelemetry.io/pull/8624/commits/86e60b57e61ee509c0a8eb123d691398ab0fa351.

gshriver committed:
Apply suggested mainframe meaning update

gshriver commented on content/en/docs/platforms/mainframe/_index.md:
```diff
@@ -0,0 +1,118 @@
+- Platforms such as **IBM z/OS® mainframes** and compatible environments that host:
+  - Transaction processing (e.g., CICS®, IMS™ and similar subsystems)
+  - Batch processing (JCL-driven jobs, schedulers)
+  - High-value systems of record (databases and files that are the “source of truth”)
```
Thank you for reviewing!  I tried to incorporate your suiggestion in commit https://github.com/open-telemetry/opentelemetry.io/pull/8624/commits/92ebbab1d1809a263367acd02521d81b08daec19.  PTAL.  Thank you!

chalin commented on content/en/docs/platforms/mainframe/_index.md:
```diff
@@ -0,0 +1,171 @@
+  - Batch processing (JCL-driven jobs, schedulers)
+  - High-value systems of record (databases and files that are the “source of
+    truth”)
+  - Large **high-volume transaction processing** and **batch workloads**.
```
It doesn't make sense to have a single top-level list item, maybe this instead?

```suggestion
> Platforms such as **IBM z/OS® mainframes** and compatible environments that host:
> 
> - Transaction processing (e.g., CICS®, IMS™ and similar subsystems)
> - Batch processing (JCL-driven jobs, schedulers)
> - High-value systems of record (databases and files that are the “source of
> - Large **high-volume transaction processing** and **batch workloads**.
```

---

> Platforms such as **IBM z/OS® mainframes** and compatible environments that host:
> 
> - Transaction processing (e.g., CICS®, IMS™ and similar subsystems)
> - Batch processing (JCL-driven jobs, schedulers)
> - High-value systems of record (databases and files that are the “source of
> - Large **high-volume transaction processing** and **batch workloads**.

chalin commented on content/en/docs/platforms/mainframe/_index.md:
```diff
@@ -0,0 +1,171 @@
+You do _not_ need prior mainframe experience, and you do not need to be a COBOL
+or z/OS expert to benefit from this section.
+
+## What we mean by “mainframe”
```
```suggestion
## What we mean by mainframe
```

chalin reviewed (changes requested):
Nice. Some comments, see inline.

rrschulze reviewed (commented):
Added an intro sentence for mainframe and a couple of changes to the wording.

rrschulze reviewed (approved):
Approving the document for the mainframe SIG under the assumption that proposed changes will make it in.

gshriver committed:
Update content/en/docs/platforms/mainframe/_index.md

Co-authored-by: Ruediger Schulze <140042958+rrschulze@users.noreply.github.com>

gshriver committed:
Apply suggestions from code review

Several content changes, additions, clarifications, and formatting changes from multiple reviewers.

Co-authored-by: Patrice Chalin <chalin@users.noreply.github.com>
Co-authored-by: Ruediger Schulze <140042958+rrschulze@users.noreply.github.com>

chalin commented on content/en/docs/platforms/mainframe.md:
And this issue @gshriver? Can you rename the page?

chalin commented:
/fix:all

otelbot[bot] committed:
Results from /fix directive

chalin commented on content/en/docs/platforms/mainframe/_index.md:
```diff
@@ -0,0 +1,180 @@
+
+Mainframes are data servers that are designed to process billions of
+transactions daily with the highest levels of security and reliability. For a
+more detailed overview, please visit:
```
```suggestion
more detailed overview, see 
```

chalin commented on content/en/docs/platforms/mainframe/_index.md:
```diff
@@ -0,0 +1,180 @@
+[What is a mainframe?](https://www.ibm.com/think/topics/mainframe) By
+“mainframe” we generally mean:
+
+IBM Z systems with the z/OS operating system that host host:
```
```suggestion
[What is a mainframe?](https://www.ibm.com/think/topics/mainframe).

 An example of a mainframe is an IBM Z system that hosts:
```

chalin commented on content/en/docs/platforms/mainframe/_index.md:
```diff
@@ -0,0 +1,180 @@
+  > - Transaction processing (e.g., CICS®, IMS™ and similar subsystems)
+  > - Batch processing (JCL-driven jobs, schedulers)
+  > - High-value systems of record (databases and files that are the “source of
+  > - Large **high-volume transaction processing** and **batch workloads**.
```
These lines just seem to be a block-quoted repeat of the previous lines:

```suggestion
```

chalin reviewed (changes requested):
The "What we mean by mainframe" section seems muddled (copy-paste error?). See inline comments.

chalin committed:
Add local cSpell:ignore words

chalin committed:
Fix broken link def by dropping it

chalin commented:
/fix:dict

otelbot[bot] committed:
Results from /fix directive

chalin reviewed (changes requested):
@gshriver - here's some feedback from Copilot, please address the comments, thx.

---

Remaining issues, excluding the file rename:

### Must fix

- **Lines 39–57: definition section is still broken.**
  - IBM link and “By ‘mainframe’…” run together: `...mainframe) By`.
  - `host host:` typo.
  - Duplicated competing definitions.
  - Accidental nested blockquote.
  - Truncated bullet: `source of` with no `truth`.
  - This section needs to be rewritten, not reformatted.

Suggested replacement:

```markdown name=content/en/docs/platforms/mainframe/_index.md
## What we mean by mainframe

In this section, “mainframe” generally refers to IBM Z systems running z/OS, or
compatible environments, that host:

- Transaction processing systems, such as CICS®, IMS™, and similar subsystems
- Batch workloads, such as JCL-driven jobs and scheduled processing
- High-value systems of record, such as databases and files that are sources of
  truth for business operations
- Large-scale, high-volume transaction processing and batch workloads

For a broader introduction, see [What is a mainframe?](https://www.ibm.com/think/topics/mainframe).
```

### Should fix

- **Line 66:** Prefer spelling out OpenTelemetry in the heading:
  - `## How mainframes show up in OpenTelemetry architectures`

- **Lines 141–155:** Status/vendor claims are still vague and uncited:
  - “available today”
  - “landscape is quickly changing”
  - “customers preferring”
  - IBM “currently in the process of including native OpenTelemetry support”
  
  Either link to concrete resources or make more neutral.

- **Line 150:** Replace `--` with an em dash or rephrase.

- **Lines 106–107:** Awkward phrasing:
  - Current: “The above need parsing…”
  - Better: “These sources often need to be parsed and mapped…”

### Already fixed since my prior review

- Long-line wrapping is mostly fixed.
- `Independent [Mainframe] Software Vendors` is fixed.
- Trailing spaces near the community links are fixed.

chalin commented:
👋 Please update this PR's branch over the latest `main`: it predates our link-checker [switch to Lychee][#10911] and can hit CI failures unrelated to your changes. For the how — including conflicts and after-update steps — see [#10990][]. Thanks for your contribution!

[#10911]: https://github.com/open-telemetry/opentelemetry.io/pull/10911
[#10990]: https://github.com/open-telemetry/opentelemetry.io/issues/10990

gshriver committed:
Update content/en/docs/platforms/mainframe/_index.md

Co-authored-by: Patrice Chalin <chalin@users.noreply.github.com>

gshriver committed:
Update content/en/docs/platforms/mainframe/_index.md

Co-authored-by: Patrice Chalin <chalin@users.noreply.github.com>

gshriver committed:
Update content/en/docs/platforms/mainframe/_index.md

Co-authored-by: Patrice Chalin <chalin@users.noreply.github.com>

chalin committed:
Merge upstream/main into add-mainframe-overview; drop obsolete refcache.json changes

chalin committed:
Rename platforms/mainframe/_index.md to mainframe.md

chalin committed:
Format mainframe.md with prettier

chalin committed:
Merge remote-tracking branch 'upstream/main' into pr-8624-update

chalin committed:
Add mainframe page links to link cache

chalin commented on content/en/docs/platforms/mainframe.md:
```diff
@@ -0,0 +1,173 @@
+- Long-lived applications and data formats
+- Strong security and compliance constraints
+
+## How mainframes show up in OTel architectures
```
Prefer the full project name in headings:

```suggestion
## How mainframes show up in OpenTelemetry architectures
```

chalin commented on content/en/docs/platforms/mainframe.md:
```diff
@@ -0,0 +1,173 @@
+  - job logs
+  - performance monitors
+  - (The above need parsing and mapping into **traces, metrics, and logs** as
+    defined by OpenTelemetry)
```
Phrase adjustment:

```suggestion
  - These sources often need to be parsed and mapped into traces, metrics,
    and logs as defined by OpenTelemetry
```

chalin commented on content/en/docs/platforms/mainframe.md:
```diff
@@ -0,0 +1,173 @@
+native OpenTelemetry support into its services.
+
+Many independent mainframe software vendors (ISVs) are also transitioning their
+observability support to favor the OpenTelemetry vendor-neutral approach.
```
The claims here are uncited and time-anchored ("available today", "quickly changing", "currently in the process"); they'll age poorly on a docs page. Unless you can link concrete resources, here's a neutral rewrite (it also merges the last three paragraphs):

```suggestion
Foundational OpenTelemetry instrumentation is available for mainframe
environments, and platform support continues to expand.

Historically, most mainframe instrumentation has been vendor supplied: many
observability backend vendors provide their own extensions or agents to
package and ship mainframe telemetry to their backends.

In response to customer demand for vendor-neutral telemetry, IBM, which
supplies the operating system and subsystem software for the most widely used
mainframe systems, and many independent mainframe software vendors (ISVs) are
adding native OpenTelemetry support to their products.
```

chalin commented on content/en/docs/platforms/mainframe.md:
```diff
@@ -0,0 +1,173 @@
+observability support to favor the OpenTelemetry vendor-neutral approach.
+
+Of course, this successful transition depends upon common terminology and
+semantics.
```
Nit: "Of course, …" dangles at the end of the section. Suggestion to tie it to the section that follows:

```suggestion
This transition depends on common terminology and semantics, which the
working group described in the next section is helping to define.
```

chalin reviewed (commented):
Thanks for the updates. I've rebased and update the link cache. See inline for copyedit suggestions. I'll apply them and await for a final second review from docs approvers. Then we can merge. Let me know if you'd like anything tweaked before then.

chalin committed:
Apply suggestions from code review

Co-authored-by: Patrice Chalin <chalin@users.noreply.github.com>

otelbot[bot] committed:
Results from /fix directive

chalin reviewed (approved):
Thanks @gshriver for your patience through the long review cycle! I've merged the remaining suggestions; flag anything you'd like phrased differently and we can adjust in a follow-up.

@svrnm PTAL: your December change request is the last gate. Reminder that your component owners item (@open-telemetry/sig-mainframe-approvers) is slated for a follow-up PR.

chalin committed:
Merge branch 'main' into add-mainframe-overview

chalin committed:
Merge remote-tracking branch 'upstream/main' into add-mainframe-overview

# Conflicts:
#	.lycheecache

chalin committed:
Refresh link cache after resolving conflicts

chalin committed:
Merge branch 'main' into add-mainframe-overview
